src/Microsoft.TestPlatform.Common/Utilities/AssemblyResolver.cs

```diff
@@ -175,9 +175,10 @@ private Assembly OnResolve(object sender, AssemblyResolveEventArgs args)
                         }
                     }
                 }
-            }
 
-            return null;
+                this.resolvedAssemblies[args.Name] = null;
```

Comment:
Add this at line:112, because there is exit path at 126 and we should operate on resolvedAssemblies inside `lock (this.resolvedAssemblies)` block.

Reply:
Its already under lock block. As per discussion, added this statement in one catch block.